dependencies/FindVMC.cmake

```diff
@@ -0,0 +1,20 @@
+# Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+# See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+# All rights not expressly granted are reserved.
+#
+# This software is distributed under the terms of the GNU General Public
+# License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+#
+# In applying this license CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+find_package(VMC NO_MODULE)
+if(NOT VMC_FOUND)
+  return()
+endif()
+
+# Promote the imported target to global visibility
+# (so we can alias it)
+set_target_properties(VMCLibrary PROPERTIES IMPORTED_GLOBAL TRUE)
```

Comment:
As the cmake files in vmc are the same as in geant4_vmc and geant3; I would expect the analogic code as in Find files for these packages:
```
set_target_properties(VMCLibrary
                      PROPERTIES INTERFACE_INCLUDE_DIRECTORIES
                                "${VMC_INCLUDE_DIRS}"
                                INTERFACE_LINK_DIRECTORIES
                                $<TARGET_FILE_DIR:vmc>)
```
I think, @aphecetche should know whether this is needed here, too.

Reply:
The thing is, these properties should already be defined: https://github.com/vmc-project/vmc/blob/master/source/CMakeLists.txt#L80-L81

Reply:
yes, if VMC is now a good cmake citizen (i.e. correctly provides its own targets in its `VMCConfigxxx.cmake`), the `FindVMC.cmake` we have in O2 should no longer be needed and could be removed.

Reply:
Indeed, and I build everything locally, simulation works. I even went up to building `O2sim` and it looks all good. Ran `full_system_test` as well

Reply:
> yes, if VMC is now a good cmake citizen (i.e. correctly provides its own targets in its `VMCConfigxxx.cmake`), the `FindVMC.cmake` we have in O2 should no longer be needed and could be removed.

But then we may also drop these settings from geant3 and geant4_vmc?

Reply:
so in case of `VMC` I did this in the past when I restructured some of the CMake. In the case of G3 and G4 VMC we have the include dirs interfaced.
Indeed, could somebody tell me why we have the lib dir for G4 VMC?
```bash
set_target_properties(geant4vmc
                      PROPERTIES INTERFACE_INCLUDE_DIRECTORIES
                                "${Geant4VMC_INCLUDE_DIRS}"
                                INTERFACE_LINK_DIRECTORIES
                                $<TARGET_FILE_DIR:geant4vmc>)
```
I reckon it's because we have **not** specified targets the `geant4vmc` target itself links against (and there might be some in that lib dir, such as the `g4root` deps.

I would say we first close this business which is very specific in the sense that it only concerns the transition to VMC standalone. After that I will do another round of CMake enhancement in the VMC packages so that they also move into the CMake village. What do you think?
Because that will be smooth transition without any breaks between recipes and dependencies.

Reply:
IMO yes we should focus on the minimal set of changes required for that task as hand, i.e. the migration to "standalone VMC". The rest can be dealt with later on of course.

Reply:
Yes, I agree. Thank you.